### PR TITLE
Robust Scaler now added to the Normalizer catalog

### PR DESCRIPTION
**Pull request by michaelgsharp**
One of the featurizers that was added in the past was the RobustScalerFeaturizer. After talking with the ML.Net team, it was decided that since this is just a normalizer it makes more sense to have it be part of the existing collection of normalizers that ML.Net already has. This PR adds that functionality. It is the equivalent of the [RobustScaler](https://scikit-learn.org/stable/modules/generated/sklearn.preprocessing.RobustScaler.html) in pandas. Once this PR goes in, we can remove the original transformer that was added.

### CODE REVIEW

File: test/Microsoft.ML.Tests/Transformers/NormalizerTests.cs [hunk 1]
```diff
@@ -975,5 +1031,152 @@ public void TestSavingNormalizerWithMultidimensionalVectorInput()
 
             Assert.True(dimensions1.SequenceEqual(dimensions2));
         }
+
+        [Fact]
+        public void TestMinHeapForMedianNormalizer()
+        {
```
Comment (harishsk) [inline test/Microsoft.ML.Tests/Transformers/NormalizerTests.cs:1037]: Instead of static values, would it be better to add tests that compute the median with and without the heap and verify that both produce the same results?

File: src/Microsoft.ML.Data/Transforms/Normalizer.cs [hunk 2]
```diff
@@ -47,14 +47,16 @@ namespace Microsoft.ML.Transforms
     /// * Log Mean Variance - Rescale each row to unit variance, optionally, zero mean based on computations in log scale.
     /// * Binning - Bucketizes the data in each row and performs a linear rescale based on the calculated bins.
     /// * Supervised Binning - Bucketize the data in each row and performas a linear rescale based on the calculated bins. The bin calculation is based on correlation of the Label column.
+    /// * Robust Scaling - Centers the data and scales based on the range of data and the quantile min and max values provided.
     ///
```
Comment (harishsk) [inline src/Microsoft.ML.Data/Transforms/Normalizer.cs:51]: suggestion: Append "This method is more robust to outliers"

**Comment by harishsk** [comment]
>     /// Check the above links for usage examples.

Please add an entry here for [NormalizeRobustScaling]

---
Refers to: src/Microsoft.ML.Data/Transforms/Normalizer.cs:75 in 00d4ce7. [](commit_id = 00d4ce7a445efab391e7856cf04016c7d8493df1, deletion_comment = False)

File: src/Microsoft.ML.Data/Transforms/Normalizer.cs [hunk 3]
```diff
@@ -109,7 +114,11 @@ internal enum NormalizationMode
             /// <summary>
             /// Bucketize and then rescale to between -1 and 1. Calculates bins based on correlation with the Label column.
             /// </summary>
-            SupervisedBinning = 4
+            SupervisedBinning = 4,
+            /// <summary>
+            /// Centers the data around 0 and then scales based on the data range and the quantile min and max values provided.
+            /// </summary>
```
Comment (harishsk) [inline src/Microsoft.ML.Data/Transforms/Normalizer.cs:120]: Should this read: "Centers the data around 0 optionally and..."

File: test/Microsoft.ML.Tests/Transformers/NormalizerTests.cs [hunk 4]
```diff
@@ -383,6 +387,45 @@ public void NormalizerParametersMultiColumnApi()
             Assert.Equal(4, doubleBinModelVector.Density.Length);
             Assert.Empty(doubleBinModelVector.Offset);
 
+            // Robust scaler
+            var robustScalerEstimator = context.Transforms.NormalizeRobustScaling(
+                                new[] {new InputOutputColumnPair("float1rbs", "float1"), new InputOutputColumnPair("float4rbs", "float4"),
```
Comment (harishsk) [inline test/Microsoft.ML.Tests/Transformers/NormalizerTests.cs:392]: The normalizer API haven't been tested thoroughly through Onnx I think. Can you please add a test in OnnxConversionTest.cs for onnx export for this and other Normalizer apis ?

File: test/Microsoft.ML.Tests/Transformers/NormalizerTests.cs (hunk 1 above)
Comment (michaelgsharp) [reply test/Microsoft.ML.Tests/Transformers/NormalizerTests.cs:1091]: Yep. This has now been added. The specific tests you highlighted are just testing the heap in general, but I have added another test that tests the median dynamically and compares with the naive LINQ implementation.

File: src/Microsoft.ML.Data/Transforms/Normalizer.cs (hunk 2 above)
Comment (michaelgsharp) [reply src/Microsoft.ML.Data/Transforms/Normalizer.cs:51]: Added.

File: src/Microsoft.ML.Data/Transforms/Normalizer.cs (hunk 3 above)
Comment (michaelgsharp) [reply src/Microsoft.ML.Data/Transforms/Normalizer.cs:121]: Added.

File: test/Microsoft.ML.Tests/Transformers/NormalizerTests.cs (hunk 4 above)
Comment (michaelgsharp) [reply test/Microsoft.ML.Tests/Transformers/NormalizerTests.cs:392]: I added in the ONNX export tests. We dont' support export for anything where `useCdf` is true, and neither of the binning normalizers support it, but the rest now have tests.

File: test/Microsoft.ML.Tests/OnnxConversionTest.cs [hunk 5]
```diff
@@ -2145,6 +2146,171 @@ public void NonDefaultColNamesMultiClassificationOnnxConversionTest()
...
+        /// converts the chain to an ONNX model and compares the results.
+        /// </summary>
+        /// <typeparam name="TLastTransformer">The type of for the EstimatorChain. Will be auto detected.</typeparam>
+        /// <param name="mlContext">The MLContext</param>
+        /// <param name="pipeline">A single IEstimator{ITransformer} to be tested.</param>
+        /// <param name="dataView">The test data.</param>
+        /// <param name="onnxFileName">Name to save the ONNX model file.</param>
+        /// <param name="columnsToCompare">Which columns you want to compare. This assumes that the column name in ONNX is the same as ML.Net, so only 1 name per column is provided.</param>
+        private void TestPipeline<TLastTransformer>(MLContext mlContext, EstimatorChain<TLastTransformer> pipeline, IDataView dataView, string onnxFileName, params string[] columnsToCompare)
+            where TLastTransformer : class, ITransformer
```
Comment (harishsk) [inline test/Microsoft.ML.Tests/OnnxConversionTest.cs:2288]: Other tests have used a new MLContext object for their tests but it is strictly not needed. The base test class already has one and that can be reused. 
So it would be good to not have MLContext as a param here and just use the object from the base class.

File: test/Microsoft.ML.Tests/OnnxConversionTest.cs [hunk 6]
```diff
@@ -2145,6 +2146,171 @@ public void NonDefaultColNamesMultiClassificationOnnxConversionTest()
...
+        /// EstimatorChain{ITransformer} and then calls the actual testing method.
+        /// </summary>
+        /// <param name="mlContext">The MLContext</param>
+        /// <param name="pipeline">A single IEstimator{ITransformer} to be tested.</param>
+        /// <param name="dataView">The test data.</param>
+        /// <param name="onnxFileName">Name to save the ONNX model file.</param>
+        /// <param name="columnsToCompare">Which columns you want to compare. This assumes that the column name in ONNX is the same as ML.Net, so only 1 name per column is provided.</param>
+        /// <param name="schemaDefinition">Optional schema definition for the IEnumerable{TRow}.</param>
+        private void TestPipeline(MLContext mlContext, IEstimator<ITransformer> pipeline, IDataView dataView, string onnxFileName, string[] columnsToCompare, SchemaDefinition schemaDefinition = null)
+        {
```
Comment (harishsk) [inline test/Microsoft.ML.Tests/OnnxConversionTest.cs:2252]: Is the SchemaDefinition parameter needed here?

File: test/Microsoft.ML.Tests/OnnxConversionTest.cs (hunk 5 above)
Comment (michaelgsharp) [reply test/Microsoft.ML.Tests/OnnxConversionTest.cs:2283]: Good point. It is now removed.

File: test/Microsoft.ML.Tests/OnnxConversionTest.cs (hunk 6 above)
Comment (michaelgsharp) [reply test/Microsoft.ML.Tests/OnnxConversionTest.cs:2248]: Nope. Its now removed.
